### PR TITLE
Test for requesting a DOI on Zenodo for a hosted workflow

### PR DESCRIPTION
**Pull request by wshands**
Will login to Dockstore, try to locate a hosted workflow of the name 'zenodo-doi-test' and if not found create it. Then will create version and request a DOI for the version.

Zenodo credentials must be linked before running it.
https://ucsc-cgl.atlassian.net/browse/DOCK-898

### CODE REVIEW

**Comment by agduncan94** [comment]
Extending what Denis said, this should go in its own file and should never be run on travis automatically.

**Comment by garyluu** [comment]
Possible to not have GitHub show my changes in this PR?

File: cypress/integration/manualTests/zenodo.ts [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+        // accounts for no colon because an empty string is falsy, therefore it returns workflowPath
+        // https://stackoverflow.com/questions/4250364/how-to-trim-a-file-extension-from-a-string-in-javascript
+        workflowName = workflowName.substring(0, workflowName.lastIndexOf(':')) || workflowName;
+
+        if (workflowName !== zenodoWorkflow) {
+          // Create the Zenodo test hosted workflow
+          cy.get('[id=registerWorkflowButton]').click();
+          // Select the Use CWL, WDL or Nextflow from GitHub, Bitbucket, etc.' option to create a hosted workflow
+          cy.contains('Create and save CWL, WDL, or Nextflow on Dockstore.org').click();
+          // TODO: Fix this.  When 'Next' is clicked too fast, the next step is empty
```
Comment (coverbeck) [inline cypress/integration/manualTests/zenodo.ts]: Does this comment still apply, or does your wait on the next line handle it?

File: cypress/integration/manualTests/zenodo.ts [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+          cy.wait(1000);
+          cy.contains('button', 'Next').click();
+          cy.get('#descriptorTypeRadioButtons').contains(wdlDescriptorType).find('.mat-radio-container').click();
+          cy.get('[id="hostedWorkflowRepository"]').type(zenodoWorkflow);
+          // TODO: Fix this.  When 'Next' is clicked too fast, the next step is empty
```
Comment (coverbeck) [inline cypress/integration/manualTests/zenodo.ts]: Same as previous comment/question.

File: cypress/integration/manualTests/zenodo.ts [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+ import { goToTab } from '../../support/commands';
+
+function testID_Alpha() {
```
Comment (coverbeck) [inline cypress/integration/manualTests/zenodo.ts]: In TypeScript, at least the way we're doing it, you don't typically have underscores in function names.

Also, I don't think the name of the function conveys its meaning very well. It's a random string generator, not a test function.

**Comment by agduncan94** [comment]
@wshands You should look into waiting for requests to end instead of waiting for fixed amounts of time. It should make the tests run faster/be less error prone.

https://docs.cypress.io/guides/guides/network-requests.html#Waiting

File: cypress/integration/manualTests/zenodo.ts (hunk 1 above)
Comment (wshands) [reply cypress/integration/manualTests/zenodo.ts]: removed waits and fixed this

File: cypress/integration/manualTests/zenodo.ts (hunk 2 above)
Comment (wshands) [reply cypress/integration/manualTests/zenodo.ts]: removed waits and fixed

**Pull request merged**